src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
+
+
+    private T[] array = (T[]) new Object[def_capacity];
+    private int size;
+    private int cursor;
+
+
+    private String errorMsg(int index) {
+        return "Index: " + index + ", Size: " + size;
+    }
+
+
+    private Object[] grow() {
+        int oldCapacity = array.length;
+        double multiplicator = 1;
+        if (oldCapacity == size) {
+            multiplicator = 1.5;
+        }
+        if (oldCapacity > 0 || array != empty_arraydata) {
+            int newCapacity = (int) (oldCapacity * multiplicator + 1);
+            return array = Arrays.copyOf(array, newCapacity);
+        } else {
+            return array = (T[]) new Object[def_capacity];
+        }
+    }
+
+
+    public static Object[] listToArray(List<?> list) {
```

Comment:
make it generic and non static
```suggestion
    public <T> T[] listToArray(List<T> list) {
```